src/pages/LabIntakeInput.js

```diff
@@ -1,13 +1,145 @@
 import React from "react";
+import { DECIMAL_REGEX } from "../constants/constants";
 
 import "../App.css";
 import "./home.css";
 
-const LabIntakeInput = ({ labState, setLabState }) => {
-  const handleInputChange = (e) => {
+const LabIntakeInput = ({
+  labState,
+  setLabState,
+  crpError,
+  setCrpError,
+  esrError,
+  setEsrError,
+  dDimerError,
+  setDdimerError,
+  ferritinError,
+  setFerritinError,
+  ldhError,
+  setLdhError,
+  wbcError,
+  setWbcError,
+  neutrophilError,
+  setNeutrophilError,
+  lymphocytesError,
+  setLymphocytesError,
+  eosinophilsError,
+  setEosinophilsError,
+  basophilsError,
+  setBasophilsError,
+  plateletsError,
+  setPlateletsError,
+}) => {
+  const handleCrpChange = (e) => {
     const item = e.target.name;
+    if (e.target.value === "" || e.target.value.match(DECIMAL_REGEX)) {
+      setCrpError(false);
+    } else {
+      setCrpError(true);
+    }
     setLabState({ ...labState, [item]: e.target.value });
   };
+
+  const handleEsrChange = (e) => {
+    const item = e.target.name;
+    if (e.target.value === "" || e.target.value.match(DECIMAL_REGEX)) {
+      setEsrError(false);
+    } else {
+      setEsrError(true);
+    }
+    setLabState({ ...labState, [item]: e.target.value });
+  };
```

Comment:
please create one function and use it for all inputs:
```suggestion
  const handleEsrChange = (e) => {
    const item = e.target.name;
    const value = e.target.value;
   
    setLabErrors(errors => ({ ...errors, [item]: value === "" || value?.match(DECIMAL_REGEX)?.[0] }));
    setLabState(state => ({ ...state, [item]: value }));
  };
```

Reply:
resolved